src/EditorFeatures/CSharpTest/CodeActions/ExtractMethod/ExtractMethodTests.cs

```diff
@@ -1310,5 +1310,232 @@ private static string NewMethod(int i)
+{
+    public static void Main(string[] args)
+    {
+        bool Local() => args == null;
+        [|Local();|]
+    }
+}", @"
+class C
+{
+    public static void Main(string[] args)
+    {
+        bool Local() => args == null;
+        {|Rename:NewMethod|}();
+    }
+
+    private static void NewMethod()
+    {
+        {|Warning:Local();|}
+    }
+}");
+        }
+
+        [Fact]
+        [WorkItem(15532, "https://github.com/dotnet/roslyn/issues/15532")]
+        public async Task ExtractLocalFunctionDeclaration()
+        {
+            await TestMissingAsync(@"
+class C
+{
+    public static void Main()
+    {
+        [|bool Local() => args == null;|]
+        Local();
+    }
+}");
+        }
+
+        [Fact]
+        [WorkItem(15532, "https://github.com/dotnet/roslyn/issues/15532")]
+        public async Task ExtractLocalFunctionInterior()
+        {
+            await TestAsync(@"
+class C
+{
+    public static void Main()
+    {
+        void Local()
+        {
+            [|int x = 0;
+            x++;|]
+        }
+        Local();
+    }
+}", @"
+class C
+{
+    public static void Main()
+    {
+        void Local()
+        {
+            {|Rename:NewMethod|}();
+        }
+        Local();
+    }
+
+    private static void NewMethod()
+    {
+        {|Warning:int x = 0;
+        x++;|}
+    }
+}");
+        }
+
+        [WorkItem(538229, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/538229")]
+        [Fact, Trait(Traits.Feature, Traits.Features.ExtractMethod)]
+        public async Task Bug3790()
+        {
+            await TestAsync(@"
+class Test
+{
+    void method()
+    {
+        static void Main(string[] args)
+        {
+            int v = 0;
+            for(int i=0 ; i<5; i++)
+            {
+                [|v = v + i;|]
+            }
+        }
+    }
+}", @"
+class Test
+{
+    void method()
+    {
+        static void Main(string[] args)
+        {
+            int v = 0;
+            for(int i=0 ; i<5; i++)
+            {
+                {|Rename:NewMethod|}();
+            }
+        }
+    }
+
+    private static void NewMethod()
+    {
+        {|Warning:v = v + i|};
+    }
+}");
+        }
+
+        [WorkItem(538229, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/538229")]
+        [Fact, Trait(Traits.Feature, Traits.Features.ExtractMethod)]
+        public async Task Bug3790_1()
+        {
+            await TestAsync(@"
+class Test
+{
+    void method()
+    {
+        static void Main(string[] args)
+        {
+            int v = 0;
+            for(int i=0 ; i<5; i++)
+            {
+                [|v = v + i|];
+            }
+        }
+    }
+}", @"
+class Test
+{
+    void method()
+    {
+        static void Main(string[] args)
+        {
+            int v = 0;
+            for(int i=0 ; i<5; i++)
+            {
+                v = {|Rename:NewMethod|}();
+            }
+        }
+    }
+
+    private static int NewMethod()
+    {
+        {|Warning:return v + i|};
```

Comment:
So does this actually work?  I just looked at our test harness and it seems busted.  Namly it does this:

```
            var annotatedTokens = fixedRoot.GetAnnotatedNodesAndTokens(annotationKind).Select(n => (SyntaxToken)n).ToList();
```
Which makes it seem like it only expects tokens to be annotated.  (note that this is a flaw in the test harness)... i'm just surprised that this is working.